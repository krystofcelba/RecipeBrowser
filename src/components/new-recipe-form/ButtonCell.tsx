import React from 'react';
import { StyleSheet } from 'react-native';
import { Cell } from 'react-native-tableview-simple';

import AppTheme from 'src/assets/appTheme';

interface Props {
  title: string;
  onPress: () => void;
}

const ButtonCell = ({ title, onPress }: Props) => (
  <Cell title={title} titleTextStyle={styles.title} onPress={onPress} />
);

const styles = StyleSheet.create({
  title: {
    color: AppTheme.secondaryColor,
    fontFamily: AppTheme.fontFamily,
    fontSize: AppTheme.defaultFontSize,
  },
});

export default ButtonCell;
