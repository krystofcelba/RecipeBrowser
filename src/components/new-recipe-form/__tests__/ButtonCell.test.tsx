import React from 'react';
import { shallow } from 'enzyme';

import ButtonCell from '../ButtonCell';

describe('FormButtonCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<ButtonCell testID="test:add" title="test" onPress={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('call onPress as expected', () => {
    const onPress = jest.fn();
    const wrapper = shallow(<ButtonCell testID="test:add" title="test" onPress={onPress} />);
    shallow(wrapper.prop('cellContentView')).simulate('press');
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
