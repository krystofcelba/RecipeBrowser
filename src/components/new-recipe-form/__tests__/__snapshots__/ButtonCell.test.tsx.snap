// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButtonCell component renders as expected 1`] = `
<Cell
  accessory={false}
  accessoryColor="#007AFF"
  allowFontScaling={true}
  backgroundColor="#FFF"
  cellAccessoryView={null}
  cellContentView={
    <glamorous.TouchableOpacity
      flex={1}
      onPress={[Function]}
      testID="test:add"
    >
      <AddButtonText>
        test
      </AddButtonText>
    </glamorous.TouchableOpacity>
  }
  cellImageView={null}
  cellStyle="Basic"
  contentContainerStyle={Object {}}
  detail=""
  detailTextStyle={Object {}}
  disableImageResize={false}
  highlightActiveOpacity={0.8}
  highlightUnderlayColor="black"
  image={null}
  isDisabled={false}
  leftDetailColor="#007AFF"
  onHighlightRow={null}
  onPress={false}
  onUnHighlightRow={null}
  rightDetailColor="#8E8E93"
  subtitleColor="#000"
  subtitleTextStyle={Object {}}
  title=""
  titleTextColor="#000"
  titleTextStyle={Object {}}
  titleTextStyleDisabled={Object {}}
/>
`;
